dracut/full-dmroot/qubes_cow_setup.sh

```diff
@@ -54,20 +54,18 @@ modprobe xenblk || modprobe xen-blkfront || warn "Qubes: Cannot load Xen Block F
 
 log_begin "Waiting for /dev/xvda* devices..."
 udevadm settle --exit-if-exists=/dev/xvda
-/usr/sbin/gptfix /dev/xvda
-log_end
 
 # prefer partition if exists
-if [ -b /dev/xvda1 ]; then
-    if [ -e "/dev/disk/by-partlabel/Root\\x20filesystem" ]; then
-        ROOT_DEV=$(readlink "/dev/disk/by-partlabel/Root\\x20filesystem")
-        ROOT_DEV=${ROOT_DEV##*/}
-    else
-        ROOT_DEV=xvda3
-    fi
+# Fix up partition tables
+if /usr/sbin/gptfix fix /dev/xvda; then
+    udevadm settle --exit-if-exists=/dev/xvda3
+    ROOT_DEV=xvda3
+elif [ "$?" -ne 8 ]; then
+    die 'Invalid partition table that cannot be fixed'
```

Comment:
I think I did asked to not interrupt the boot if the tool doesn't know what to do...